aviatrix/resource_aviatrix_account.go

```diff
@@ -195,23 +209,29 @@ func resourceAviatrixAccountCreate(d *schema.ResourceData, meta interface{}) err
 		}
 
 		log.Printf("[INFO] Creating Aviatrix account: %#v", account)
-		if aws_iam := d.Get("aws_iam").(bool); aws_iam {
-			var role_app bytes.Buffer
-			var role_ec2 bytes.Buffer
-			role_app.WriteString("arn:aws:iam::")
-			role_app.WriteString(account.AwsAccountNumber)
-			role_app.WriteString(":role/aviatrix-role-app")
-			role_ec2.WriteString("arn:aws:iam::")
-			role_ec2.WriteString(account.AwsAccountNumber)
-			role_ec2.WriteString(":role/aviatrix-role-ec2")
-			if aws_role_app := d.Get("aws_role_app").(string); aws_role_app == "" {
-				account.AwsRoleApp += role_app.String()
+		if awsIam {
+			if _, ok := d.GetOk("aws_role_app"); !ok {
+				var roleApp bytes.Buffer
+				roleApp.WriteString("arn:aws:iam::")
+				roleApp.WriteString(account.AwsAccountNumber)
+				roleApp.WriteString(":role/aviatrix-role-app")
```

Comment:
Nitpick, don't have to change: I know you didn't write this originally, but shouldn't this just be a fmt.Sprintf ? not sure why we use a byte buffer for such a small string

Reply:
This makes sense. I changed it to fmt.Sprintf.